app/presenters/api_docs/api_reference.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module ApiDocs
+  class ApiReference
+    attr_reader :document
+
+    delegate :servers, to: :document
+
+    def initialize(spec)
+      @document = Openapi3Parser.load(spec)
+    end
+
+    def operations
+      http_operations = document.paths.flat_map do |path_name, path|
+        %w[get put post delete patch].map do |http_verb|
+          operation = path.public_send(http_verb)
+          next unless operation.is_a?(Openapi3Parser::Node::Operation)
+
+          ApiDocs::ApiOperation.new(http_verb: http_verb, path_name: path_name, operation: operation)
+        end
+      end
+
+      http_operations.compact
+    end
+
+    def schemas
+      document.components.schemas.values.map do |schema|
+        ApiDocs::ApiSchema.new(schema)
+      end
+    end
+
+    def field_lengths_summary
+      rows = flatten_hash(LeadProviderApiSpecification.as_hash)
+
+      rows.reduce([]) do |arr, (field, length)|
+        if field.include?("Length")
+          arr << [field.gsub("components.schemas.", ""), length]
+        else
+          arr
+        end
+      end
+    end
+
+  private
+
+    def flatten_hash(hash)
+      hash.each_with_object({}) do |(k, v), h|
+        if v.is_a? Hash
+          flatten_hash(v).map do |h_k, h_v|
+            h["#{k}.#{h_k}"] = h_v
+          end
+        else
+          h[k] = v
+        end
```

Comment:
I hate one-letter variables, and the only thing I hate more is three letter variables which have two words in them - both in the form of single letters 😂 

Could we name them better? 

```
      hash.each_with_object({}) do |(key, value), result_hash|
        if value.is_a? Hash
          flatten_hash(value).each do |nested_hash_key, nested_hash_value|
            result_hash["#{key}.#{nested_hash_key}"] = new_hash_value
          end
        else
          result_hash[key] = value
        end
```
you will notice I replaced `map` with `each` - I think we are not actually using the fact that `map` generates a new array in that function. Which would maybe be easier to catch if variables had longer names :)